packages/workflows/src/definition/product/create-product-variants.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  TransactionStepsDefinition,
+  WorkflowManager,
+} from "@medusajs/orchestration"
+import { InputAlias, Workflows } from "../../definitions"
+import { exportWorkflow, pipe } from "../../helper"
+
+import { ProductTypes, WorkflowTypes } from "@medusajs/types"
+import { ProductHandlers } from "../../handlers"
+
+export enum CreateProductVariantsActions {
+  prepare = "prepare",
+  createProductVariants = "createProductVariants",
+  revertProductVariantsCreate = "revertProductVariantsCreate",
+  upsertPrices = "upsertPrices",
+}
+
+export const workflowSteps: TransactionStepsDefinition = {
+  next: {
+    action: CreateProductVariantsActions.prepare,
+    noCompensation: true,
+    next: {
+      action: CreateProductVariantsActions.createProductVariants,
+      noCompensation: true,
+      next: [
+        {
+          action: CreateProductVariantsActions.upsertPrices,
+        },
+      ],
+    },
+  },
+}
+
+const handlers = new Map([
+  [
+    CreateProductVariantsActions.prepare,
+    {
+      invoke: pipe(
+        {
+          merge: true,
+          inputAlias: InputAlias.ProductVariantsCreateInputData,
+          invoke: {
+            from: InputAlias.ProductVariantsCreateInputData,
+          },
+        },
+        ProductHandlers.createProductVariantsPrepareData
+      ),
+    },
+  ],
+  [
+    CreateProductVariantsActions.createProductVariants,
+    {
+      invoke: pipe(
+        {
+          merge: true,
+          invoke: {
+            from: CreateProductVariantsActions.prepare,
+          },
+        },
+        ProductHandlers.createProductVariants
```

Comment:
todo: should we add a `compensate` here? If upserting prices in the next step fails we should delete the variant.
Also, it would be good to add an integration for testing compensation if something fails in this workflow.

Reply:
added compensation